datafusion/core/src/physical_plan/aggregates/hash.rs

```diff
@@ -326,10 +402,20 @@ fn group_aggregate_batch(
                         )
                     })
                     .try_for_each(|(accumulator, values)| match mode {
-                        AggregateMode::Partial => accumulator.update_batch(&values),
+                        AggregateMode::Partial => {
+                            let size_pre = accumulator.size();
```

Comment:
👍

Reply:
You might also consider pulling the size accounting to before/after the `match` to avoid the duplication

Reply:
done